drake/systems/plants/@Manipulator/parameterEstimation.m

```diff
@@ -80,137 +124,140 @@
+    elseif isEnergetic
+        % If need memory efficiency, can cut down on # of msspoly
+        % this is just a bit more readable
+        % Initialize msspoly variables for expressing dE = E(t_2)-E(t_1)
+        dt=msspoly('dt',1);
+        
+        q1=msspoly('qo',nq);
+        s1=msspoly('so',nq);
+        c1=msspoly('co',nq);
+        qt1=TrigPoly(q1,s1,c1);
+        qd1=msspoly('qdto',nq);
+        
+        q2=msspoly('qf',nq);
+        s2=msspoly('sf',nq);
+        c2=msspoly('cf',nq);
+        qt2=TrigPoly(q2,s2,c2);
+        qd2=msspoly('qdtf',nq);
+        
+        % Formulate energy equations
+        [T1,U1] = energy(pobj,[qt1;qd1]);
+        [T2,U2] = energy(pobj,[qt2;qd2]);
+        
+        % ACROBOT-SPECIFIC FORMULATION - MUST CHANGE
+        % Need to formulate energy dissipation from AcrobotPlant class
+        dE = (B*u-[p(1);p(2)].*qd1)'*qd1*dt;
+        err = (T1+U1)-(T2+U2)+dE;
+        
+        % % DEBUGGING the energy model
+        % E1 = getmsspoly(T1+U1);
+        % E = subs(E1,p,p_orig);
+        % E_data = msubs(E,[q1;s1;c1;qd1;u;dt],[q_data;s_data;c_data;qd_data;u_data;t_data]);
+        % diffE_data = diff(E_data);
+        % dE1 = subs(dE,p,p_orig);
+        % dE_data = msubs(dE1,[q1;qd1;q2;qd2;u;dt],[q_data(:,1:end-1);qd_data(:,1:end-1);...
+        %     q_data(:,2:end);qd_data(:,2:end);u_data(:,1:end-1);dt_data]);
+        % figure;plot(diffE_data,'-g'); hold on; plot(dE_data,'-r');
+        % title(['True \DeltaE vs. Estimated \DeltaE']);
+        % xlabel('Sample Time')
+        % ylabel({'\DeltaE Magnitude' '(Joules)/sample'})
+        % legend('True \DeltaE','Estimated \DeltaE');
+    end
+    % Isolate parameters from error equations
+    [lp,M,Mb,lin_params,beta] = identifiableParameters(getmsspoly(err),p); % posynomial lumped params
+    % [lp, M, Mb] = linearParameters(getmsspoly(err),p); % monomial lumped params
+    nlp = length(lp);
+    lp_orig = double(subs(lp,p,p_orig));
+    lumped_params = msspoly('lp',nlp); 
+    % now err=M*lp+Mb and lperr=M*lumped_params+Mb;
+    
+    if isDynamic
+        ndata = length(t_data);
+        variables = [q;s;c;qd;qdd;u];
+        data_variables = [q_data;s_data;c_data;qd_data;qdd_data;u_data];
+        M_data = reshape(msubs(M(:),variables,data_variables)',nq*ndata,nlp);
+        Mb_data = reshape(msubs(Mb,variables,data_variables)',nq*ndata,1);
+    elseif isEnergetic
+        variables = [q1;s1;c1;qd1;q2;s2;c2;qd2;u;dt];
+        data_variables = [q_data(:,1:end-1);s_data(:,1:end-1);c_data(:,1:end-1);qd_data(:,1:end-1);...
+            q_data(:,2:end);s_data(:,2:end);c_data(:,2:end);qd_data(:,2:end);u_data(:,1:end-1);dt_data];
+        M_data = msubs(M(:),variables,data_variables)';
+        Mb_data = msubs(Mb,variables,data_variables)';
+    end
+elseif strcmp(options.model,'simerr')
+    % Using MATLAB sym because need to represent rational functions
+    % Perhaps a better way?
+    [xdot,dxdot,ps,qs,qds,us,ts] = pobj.dynamicsSym([qt;qd],u,t);
```

Comment:
Actually, I think I figured out how to make symbolic variables efficient enough.  I used the sym variables to create Matlab functions of the jacobians wrt the parameters and wrt the state variables - which is done only once before calling the solver.  After that, we actually don't use syms any more.  The Matlab function will evaluate the jacobians just as fast as writing the expression by hand.
My biggest concern, and why I agree that in the long run symbolic variables is a no-go, is that the syms code is not open source and we'd eventually have to write our own code C++, if we choose to move away from Matlab in the future.
I didn't realize TrigPolySystem could handle rational polynomials.  Sorry, I was just trying to make a working version so I could verify that this method is viable.  I will look into making it work with that.